plugins/languages/treesitter/treesitter.nix

```diff
@@ -6,210 +6,235 @@
+
+  maintainers = [ ];
+
+  optionsRenamedToSettings = [
+    "ensureInstalled"
+    "ignoreInstall"
+    "indent"
+    "parserInstallDir"
+    [
+      "incrementalSelection"
+      "enable"
+    ]
+    [
+      "incrementalSelection"
+      "keymaps"
+      "initSelection"
+      "nodeDecremental"
+    ]
+    [
+      "incrementalSelection"
+      "keymaps"
+      "initSelection"
+      "nodeIncremental"
+    ]
+    [
+      "incrementalSelection"
+      "keymaps"
+      "initSelection"
+      "scopeIncremental"
+    ]
+  ];
+
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "treesitter"
+      ];
+      settingsPath = basePluginPath ++ [ "settings" ];
+    in
+    [
+      (lib.mkRenamedOptionModule (basePluginPath ++ [ "customCaptures" ]) (
+        settingsPath
+        ++ [
+          "highlight"
+          "custom_captures"
+        ]
+      ))
+      (lib.mkRenamedOptionModule (basePluginPath ++ [ "disabledLanguages" ]) (
+        settingsPath
+        ++ [
+          "highlight"
+          "disable"
+        ]
+      ))
+    ];
+
+  settingsOptions = {
+    additional_vim_regex_highlighting = helpers.defaultNullOpts.mkBool false ''
+      Setting this to true will run `:h syntax` and tree-sitter at the same time.
+      Set this to `true` if you depend on 'syntax' being enabled (like for indentation).
+      Using this option may slow down your editor, and you may see some duplicate highlights.
+      Instead of true it can also be a list of languages
+    '';
+
+    highlight = {
+      enable = helpers.defaultNullOpts.mkBool true ''
+        Whether to enable treesitter highlighting.
+      '';
+
+      disable = helpers.defaultNullOpts.mkStrLuaFnOr (types.listOf types.str) "" ''
+        Can either be a list of the names of parsers you wish to disable or
+        a lua function that returns a boolean indicating the parser should be disabled.
+      '';
+
+      custom_captures = helpers.defaultNullOpts.mkAttrsOf types.str { } ''
+        Custom capture group highlighting.
+      '';
+    };
 
-      ensureInstalled = mkOption {
-        type =
-          with types;
-          oneOf [
-            (enum [ "all" ])
-            (listOf str)
-          ];
-        default = "all";
-        description = "Either \"all\" or a list of languages";
-      };
+    incremental_selection = {
+      enable = helpers.defaultNullOpts.mkBool true ''
+        Incremental selection based on the named nodes from the grammar.
+      '';
 
-      gccPackage = helpers.mkPackageOption {
-        default = if cfg.nixGrammars then null else pkgs.gcc;
-        description = ''
-          Which package (if any) to be added as the GCC compiler.
-          This is required to build grammars if you are not using `nixGrammars`.
-          To disable the installation of GCC, set this option to `null`.
-        '';
-      };
+      keymaps =
+        let
+          keymap = default: helpers.defaultNullOpts.mkStr default ''Key shortcut'';
```

Comment:
```suggestion
          mkKeymap = default: helpers.defaultNullOpts.mkStr default "Key shortcut";
```

Reply:
We should double check that upstream only accepts string values; some plugins accept either strings or tables for keymaps

Reply:
If it's likely that other keymaps get added upstream, would it be simpler to just make an `attrsOf` keymaps option instead of defining individual keymap options?

Reply:
> We should double check that upstream only accepts string values; some plugins accept either strings or tables for keymaps

https://github.com/nvim-treesitter/nvim-treesitter/blob/a8c082b4f5b3b70bde2ffb08cc112dfd7e515fb3/lua/nvim-treesitter/incremental_selection.lua#L138 looks like it is just the string. It uses the string you pass in with the function name to create the keymap.